File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict = {}
    
    for p in participant:
        dict[p] = dict.get(p, 0) + 1
    
    for c in completion:
        dict[c] -= 1
        
    for key, value in dict.items():
        if value == 1:
            return key

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        HashMap<String, Integer> nameMap = new HashMap<>();
        
        for (String p : participant) {
            nameMap.put(p, nameMap.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            if (nameMap.containsKey(c)) {
                nameMap.replace(c, nameMap.get(c) - 1);
            }
        }
        
        for (Map.Entry<String, Integer> entry : nameMap.entrySet()) {
            if (entry.getValue() > 0) {
                return entry.getKey();
            }
        }
        
        return "";
    }
}